### ISSUE TITLE
Experiment: 1000-epoch training with learned rounding + custom embeddings

### ISSUE THREAD

**Issue by LiamConnell** [LiamConnell/TinyDiffusionModels#15]
# Text Diffusion Training Experiment - 1000 Epochs

## Experiment Configuration
- **Training Duration**: 1000 epochs
- **Architecture**: Learned rounding function + custom embedding space
- **Model Features**: 
  - Custom learned embeddings (initialized from pre-trained)
  - Trainable rounding function replacing cosine similarity
  - Joint optimization of diffusion, rounding, and embedding components

## Deployment Details
- **Job Type**: `shakespeare-training`
- **Machine Type**: Standard GPU configuration  
- **Commit**: `88dd9a5` (impl_diffLM_2022 branch)
- **Expected Runtime**: ~20-30 hours

## Objectives
- Test new Diffusion-LM inspired architecture on extended training duration
- Evaluate long-term convergence and potential overfitting
- Assess quality improvements from extended joint optimization
- Validate stability of three-component training over many epochs

## Monitoring Commands
```bash
# Monitor job status
uv run python deployment/monitor.py JOB_ID

# View training logs  
uv run python deployment/monitor.py JOB_ID --logs

# Download checkpoint when complete
gsutil cp gs://text-diffusion/diffusion/checkpoints/text-model.pth ./
```

## Expected Outcomes
- Successful extended training with stable loss convergence
- High-quality checkpoint with well-optimized embedding space
- Superior text generation quality compared to shorter training runs
- Insights into optimal training duration for joint optimization

## Research Questions
- Does extended training improve embedding space quality significantly?
- How do the three loss components (diffusion, rounding, embedding) balance over time?
- What is the optimal training duration for this architecture?

## Status
- [ ] Training job submitted
- [ ] Job monitoring active  
- [ ] Periodic checkpoint evaluation
- [ ] Training completion confirmed
- [ ] Final checkpoint downloaded and tested
- [ ] Generation quality comparison with 100-epoch run
